src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,48 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            if ((getWidth() == null && rectangle.getWidth() != null)
+                    || (getWidth() != null && rectangle.getWidth() == null)
+                    || (getLength() == null && rectangle.getWidth() != null)
+                    || (getLength() != null && rectangle.getWidth() == null)
+                    || (getColor() == null && rectangle.getColor() != null)
+                    || (getColor() != null && rectangle.getColor() == null)) {
+                return false;
+            }
+
+            return (((getWidth() == null && rectangle.getWidth() == null)
+                    || getWidth().equals(rectangle.getWidth()))
+                    && ((getLength() == null && rectangle.getWidth() == null)
+                    || getLength().equals(rectangle.getLength()))
+                    && ((getColor() == null && rectangle.getColor() == null)
+                    || (getColor().equals(rectangle.getColor()))));
```

Comment:
move all this logic to a single return statement please
see checklist: While comparing fields at equals, use single return statement for all fields(e.g. return check1 && check2 && check3;)

also, don't use getters - use fields instead. you have direct access to the fields from the class where they are declared